### PR TITLE
[wip] gossip

### PR DESCRIPTION
**Pull request by marie-fourier**

### CODE REVIEW

File: packages/cli/src/options/bundlerOptions/network.ts [hunk 1]
```diff
@@ -6,13 +6,15 @@ export interface INetworkArgs {
...
   "p2p.bootEnrs": string[];
+  "p2p.dataDir": string;
 }
 
 export function parseArgs(args: INetworkArgs): IBundlerOptions["p2p"] {
   return {
     host: args["p2p.host"],
     port: args["p2p.port"],
     bootEnrs: args["p2p.bootEnrs"],
+    dataDir: args["p2p.dataDir"],
```
Comment (ch4r10t33r) [inline packages/cli/src/options/bundlerOptions/network.ts:17]: the dataDir should be the same for both. We don't need a separate configuration setting for this. Please re-use the earlier dataDir parameter.

File: packages/executor/src/services/MempoolService.ts [hunk 2]
```diff
@@ -27,6 +31,27 @@ export class MempoolService {
   }
 
   async dump(): Promise<MempoolEntrySerialized[]> {
+    await this.nodeApi.publishUserOpWithEntryPoint({
```
Comment (ch4r10t33r) [inline packages/executor/src/services/MempoolService.ts:34]: Why do we have hardcoded values here?

File: packages/node/src/network/gossip/topic.ts [hunk 3]
```diff
@@ -58,7 +63,7 @@ function stringifyGossipTopicType(topic: GossipTopic): string {
 export function getGossipSSZType(topic: GossipTopic) {
   switch (topic.type) {
     case GossipType.user_operations_with_entrypoint:
-      return ssz.UserOp;
+      return ssz.UserOpWithEntryPoint;
```
Comment (ch4r10t33r) [inline packages/node/src/network/gossip/topic.ts:66]: Please rename to `UserOpsWithEntryPoint` (plural)

File: packages/node/src/network/network.ts [hunk 4]
```diff
@@ -192,29 +180,66 @@ export class Network implements INetwork {
...
-  hasSomeConnectedPeer(): boolean {
-    return false;
+
+  getConnectedPeerCount(): number {
+    return this.peerManager.getConnectedPeerIds().length;
   }
 
   /* List of p2p functions supported by Bundler */
-  async publishUserOp(_userOp: ts.UserOp): Promise<void> {}
+  async publishUserOpWithEntryPoint(
```
Comment (ch4r10t33r) [inline packages/node/src/network/network.ts:216]: Pls rename the method to be plural `publishUserOpsWithEntryPoint`

File: packages/node/src/network/nodejs/noise.ts [hunk 5]
```diff
@@ -37,5 +37,5 @@ const lodestarCrypto: ICryptoInterface = {
 };
 
 export function createNoise(): () => ConnectionEncrypter {
-  return noise({ crypto: lodestarCrypto });
+  return noise({ crypto: crypto });
```
Comment (ch4r10t33r) [inline packages/node/src/network/nodejs/noise.ts:40]: the variable can be called varCrypto or skandhaCrypto?

**Review by ch4r10t33r — CHANGES_REQUESTED** [review]

File: packages/cli/src/index.ts [hunk 6]
```diff
@@ -24,6 +24,7 @@ void bundler
         : msg || "Unknown Error";
 
     // eslint-disable-next-line no-console
+    console.log(`STACK: ${JSON.stringify(err, undefined, 2)}`);
```
Comment (ch4r10t33r) [inline packages/cli/src/index.ts:27]: Pls replace with `logger.info` or `logger.debug`

**Review by ch4r10t33r — CHANGES_REQUESTED** [review]

File: packages/cli/src/options/bundlerOptions/network.ts (hunk 1 above)
Comment (marie-fourier) [reply packages/cli/src/options/bundlerOptions/network.ts:17]: I'll remove it later, I need this temporarily for testing

File: packages/executor/src/services/MempoolService.ts (hunk 2 above)
Comment (marie-fourier) [reply packages/executor/src/services/MempoolService.ts:34]: it's just for testing

File: packages/node/src/network/network.ts (hunk 4 above)
Comment (marie-fourier) [reply packages/node/src/network/network.ts]: done

**Review by ch4r10t33r — APPROVED** [review]

**Pull request merged**